src/Compilers/CSharp/Portable/Binder/Binder_InterpolatedString.cs

```diff
@@ -29,13 +29,22 @@ private BoundExpression BindInterpolatedString(InterpolatedStringExpressionSynta
                     case SyntaxKind.Interpolation:
                         {
                             var interpolation = (InterpolationSyntax)content;
-                            var value = BindRValueWithoutTargetType(interpolation.Expression, diagnostics);
+                            var value = BindValue(interpolation.Expression, diagnostics, BindValueKind.RValue);
+                            if (value.Type is null)
+                            {
+                                value = GenerateConversionForAssignment(objectType, value, diagnostics);
+                            }
+                            else
+                            {
+                                value = BindToNaturalType(value, diagnostics);
+                                _ = GenerateConversionForAssignment(objectType, value, diagnostics);
```

Comment:
>_ [](start = 32, length = 1)

Does this do anything besides produce diagnostics for when we can't convert to 'object'?

Reply:
This only produces diagnostics. I'm not sure which, but that's an existing scenario (so I didn't investigate deeply).

---
In reply to: [313659861](https://github.com/dotnet/roslyn/pull/37596#discussion_r313659861) [](ancestors = 313659861)